### ISSUE TITLE
Default native filters values don't set correctly

### ISSUE THREAD

**Issue by SwarrenB** [apache/superset#29186]
### Bug description

If two different users open the same dashboard and one of them changes the default value of some native filter, and then the second user closes and reopens that dashboard, the filter will be empty instead of the new default value.

### How to reproduce the bug

1. Log in as common user
2. Open some dashboard with native filters 
3. Select any values ​​in filters
4. Click "Apply Filters" button
5. Log in as admin (or another user with the ability to change filter configuration) in the new window
6. Open the same dashboard 
7. Change default value for one native filter
8. Close and reopen this dashboard as the first user again

### Superset version

3.1.3

### Python version

3.9

### Node version

16

### Browser

Chrome

### Checklist

- [X] I have searched Superset docs and Slack and didn't find a solution to my problem.
- [X] I have searched the GitHub issue tracker and didn't find a similar bug report.
- [X] I have checked Superset's logs for errors and if I found a relevant Python stacktrace, I included it here as text in the "additional context" section.

**Comment by rusackas** [comment]
Right... but personally, I don't see this as a bug. When two users open a dashboard (with or without default filter values in place), they're given a `native_filters_key` which you can see in the URL. Essentially, their state of filters is stored, and they can adjust these as they see fit without affecting other users. 

If a default filter value is then added, the other users of that dashboard are already off to the races with their "fork" of the filters values. If we made it so that adding a default value nukes everyone's filter state, that would be problematic. I'm sure there's some way to have the setting of a default value go through all users' filter settings and apply that default wherever some users have that particular filter to empty.... but that might cause other issues too (like if they _want_ it to be empty). 

We're open to proposals here, but I think that the current implementation works, and the alternatives are a bit convoluted.

**Comment by rusackas** [comment]
Closing this since we haven't heard any thoughts on the above in 9 months. Happy to revisit/reopen if you indeed think this is still a bug, or have any intentions/approaches to resolve it.
